### ISSUE TITLE
Shadow options

### ISSUE THREAD

**Issue by videosmith** [Cuperino/QPrompt#201]
Shadow appears to be limited to displaying on a transparent background.

**Describe the solution you'd prefer**
Colored shadow options on opaque background for a text highlighting option

Qprompt 1.2 rev 6 Deb/Ubuntu, x86/Arm releases

**Comment by Cuperino** [comment]
I thought the same thing during my testing. It can be almost impossible to notice with the default background, because the shadow makes things darker. The default background has a value of #0C0C0C, which is very close to black.

If one sets a brighter color, or an image as the background, the shadows become more easily noticeable. One alternative would be to make the shadows gray instead of black, but that could lead to an unnatural look and might be more computationally expensive, and may lead to shadow imperfections to being more easily noticed.

To keep performance high, shadows blurred using a variation of the "box blur" algorithm, also known as "fast blur".

**Comment by videosmith** [comment]
How about multiple instances of the same text, offset from each other?That way we could change colors in one instance, possible apply filters, etc.Or would that be expensive?
    On Monday, July 31, 2023 at 04:56:25 PM EDT, Javier O. Cordero Pérez ***@***.***> wrote:  
 
 


I thought the same thing during my testing. It can be almost impossible to notice with the default background, because the shadow makes things darker. The default background has a value of #0C0C0C, which is very close to black.

If one sets a brighter color, or an image as the background, the shadows become more easily noticeable. One alternative would be to make the shadows gray instead of black, but that could lead to an unnatural look and might be more computationally expensive, and may lead to shadow imperfections to being more easily noticed.

To keep performance high, shadows blurred using a variation of the "box blur" algorithm, also known as "fast blur".

**Comment by Cuperino** [comment]
I don't think it would be very expensive. What makes GPU interactions more expensive is the number of interactions there need to be between CPU and GPU. The less parameters need to be re-evaluated/updated, the faster the interaction.

The other thing that can make these interactions expensive is how the images are modified. Logic is evaluated in parallel on a per-pixel and per vertex level. More instances of text would increase memory consumption, but there are many shader techniques that I'm not aware of that can create illusions that achieve similar effects while consuming less resources. The effect, as it is, already creates 3 or 4 copies of the text with increasing levels of blur.
